diff --git a/gtweak/tweaks/tweak_group_appearance.py b/gtweak/tweaks/tweak_group_appearance.py
index f3c276c..4ecde6c 100644
--- a/gtweak/tweaks/tweak_group_appearance.py
+++ b/gtweak/tweaks/tweak_group_appearance.py
@@ -24,8 +24,6 @@ _shell_loaded = _shell is not None
 
 class GtkThemeSwitcher(GSettingsTweakComboRow):
     def __init__(self, **options):
-        self._gtksettings3 = GtkSettingsManager('3.0')
-        self._gtksettings4 = GtkSettingsManager('4.0')
 
         GSettingsTweakComboRow.__init__(self,
 			_("Legacy Applications"),
@@ -52,15 +50,6 @@ class GtkThemeSwitcher(GSettingsTweakComboRow):
         if item:
             value = item.value
             self.settings.set_string(self.key_name, value)
-        # Turn off Global Dark Theme when theme is changed.
-        # https://bugzilla.gnome.org/783666
-        try:
-            self._gtksettings3.set_integer("gtk-application-prefer-dark-theme",
-                                          0)
-            self._gtksettings4.set_integer("gtk-application-prefer-dark-theme",
-                                          0)
-        except:
-            self.notify_information(_("Error writing setting"))
 
 
 class IconThemeSwitcher(GSettingsTweakComboRow):
