diff --git a/src/main.rs b/src/main.rs
index da6750a..b170743 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -37,6 +37,7 @@ use std::{
     },
     panic::{self, AssertUnwindSafe},
     path::{Path, PathBuf},
+    time::{Duration, Instant},
 };
 use udev::MonitorBuilder;
 
@@ -847,6 +848,8 @@ fn real_main(drm: &mut DrmBackend) {
     let mut cfg_mgr = ConfigManager::new();
     let (mut cfg, mut layers) = cfg_mgr.load_config(width);
     let mut pixel_shift = PixelShiftManager::new();
+    let mut switch_layer = false;
+    let mut last = Instant::now();
 
     // drop privileges to input and video group
     let groups = ["input", "video"];
@@ -993,9 +996,15 @@ fn real_main(drm: &mut DrmBackend) {
                 }
                 Event::Keyboard(KeyboardEvent::Key(key)) => {
                     if key.key() == Key::Fn as u32 {
+                        if key.key_state() == KeyState::Pressed {
+                            if last.elapsed() < Duration::from_millis(200) {
+                                switch_layer = !switch_layer;
+                            }
+                            last = Instant::now();
+                        }
                         let new_layer = match key.key_state() {
-                            KeyState::Pressed => 1,
-                            KeyState::Released => 0,
+                            KeyState::Pressed => !switch_layer as usize,
+                            KeyState::Released => switch_layer as usize,
                         };
                         if active_layer != new_layer {
                             active_layer = new_layer;
